File: server/utils/email.js
const { RESEND_API_KEY, CLIENT_URL } = process.env;

let resendClient = null;

/**
 * Lazy-load Resend client — only initializes if API key is configured.
 * This allows the app to run without Resend in development/demo mode.
 */
function getResend() {
  if (resendClient) return resendClient;
  if (!RESEND_API_KEY) {
    console.warn('⚠️  RESEND_API_KEY not configured. Emails will not be sent (dev mode).');
    return null;
  }
  try {
    const { Resend } = require('resend');
    resendClient = new Resend(RESEND_API_KEY);
    return resendClient;
  } catch (err) {
    console.error('❌ Failed to initialize Resend:', err.message);
    return null;
  }
}

/**
 * Send a transactional email via Resend.
 *
 * @param {object} options
 * @param {string} options.to - Recipient email
 * @param {string} options.subject - Email subject
 * @param {string} options.html - HTML body content
 * @param {string} [options.from] - Sender (defaults to Erq Marketplace)
 * @returns {Promise<{success: boolean, data?: any, error?: string, skipped?: boolean}>}
 */
async function sendEmail({ to, subject, html, from }) {
  const resend = getResend();
  if (!resend) {
    console.log(`📧 [DEV MODE] Would send to ${to}: ${subject}`);
    return { success: false, error: 'Resend not configured', skipped: true };
  }

  try {
    const sender = from || 'Erq Marketplace <noreply@erq.et>';
    const { data, error } = await resend.emails.send({
      from: sender,
      to,
      subject,
      html,
    });

    if (error) {
      console.error('❌ Resend error:', error);
      return { success: false, error: error.message };
    }

    console.log(`✅ Email sent to ${to}: ${subject} (id: ${data?.id})`);
    return { success: true, data };
  } catch (err) {
    console.error('❌ Failed to send email:', err.message);
    return { success: false, error: err.message };
  }
}

module.exports = { sendEmail, getResend };

File: python_backend/utils/email.py
"""Email sending utility using Resend API."""

import os
import httpx
from python_backend.config import settings


async def send_email(
    to: str,
    subject: str,
    html: str,
    from_addr: str = "",
) -> dict:
    """Send a transactional email via Resend API."""
    api_key = settings.RESEND_API_KEY
    if not from_addr:
        from_addr = settings.RESEND_FROM
    if not api_key:
        print(f"[email][DEV MODE] Would send to {to}: {subject}")
        return {"success": False, "error": "Resend not configured", "skipped": True}

    try:
        async with httpx.AsyncClient(timeout=15) as client:
            response = await client.post(
                "https://api.resend.com/emails",
                headers={
                    "Authorization": f"Bearer {api_key}",
                    "Content-Type": "application/json",
                },
                json={
                    "from": from_addr,
                    "to": to,
                    "subject": subject,
                    "html": html,
                },
            )
            data = response.json()
            if response.is_error:
                print(f"[ERR] Resend error: {data}")
                return {"success": False, "error": str(data)}
            print(f"[OK] Email sent to {to}: {subject} (id: {data.get('id')})")
            return {"success": True, "data": data}
    except Exception as exc:
        print(f"[ERR] Failed to send email: {exc}")
        return {"success": False, "error": str(exc)}
